### DISCUSSION TITLE
Last downloaded beatmap must be directly selected

### DISCUSSION THREAD

**Discussion by Biokax** [ppy/osu#17365] [Ideas] [answered]
when downloading a new beatmap it doesn't select like in stable. In the website song browse we have a button to select the beatmap if it have been downloaded but it could be selected or at least add an option to make that possible

**Comment by bdach** [comment]
i don't understand. you can click the import notification after a beatmap has imported to go to it. or click the "go to" button in the beatmap listing after it has imported.

> In the website song browse we have a button to select the beatmap if it have been downloaded

i have no idea what this means. can you explain?

**Reply by Biokax** [reply]
>you can click the import notification after a beatmap has imported to go to it. or click the "go to" button in the beatmap listing after it has imported.

I didn't know about this, its really cool

>i have no idea what this means. can you explain?

I was talking about the "go to beatmap" button after you download a map
![image](https://user-images.githubusercontent.com/50368878/159165700-7e0ed651-3599-4c15-a911-141681014a03.png)


Im just not really fan of having to clic on the import notification to access the new downloaded beatmap

**Reply by bdach** [reply]
well there are different use cases. some users may prefer to download many maps at once and only then go to one of them. immediately having the beatmap be presented after download will not be a good change for them.

and making this a setting is not what i would want to see because i push back against every single new setting - the settings menu is arguably too bloated as is already.

**Reply by peppy** [reply]
I'm still a touch lost here. @Biokax you are asking for a button but the button is already there on the panel.

Are you referring to the behaviour where the last import is automatically selected on entering song select, perhaps?

**Reply by Biokax** [reply]
>Are you referring to the behaviour where the last import is automatically selected on entering song select, perhaps?

yes, I was talking about that behaviour instead of having to click the import notification

**Reply by Moltanicaa** [reply]
If you are importing many maps, you can choose to filter by "recently added" in the map selection menu.

**Reply by Biokax** [reply]
> If you are importing many maps, you can choose to filter by "recently added" in the map selection menu.

That what i've done to finally find the map, im not really fan of that

**Comment by Walavouchey** [comment]
> well there are different use cases. some users may prefer to download many maps at once and only then go to one of them. immediately having the beatmap be presented after download will not be a good change for them.

What if the game switched to the last downloaded beatmap, but didn't close the beatmap listing overlay? That sounds like the best of both worlds to me

**Comment by peppy** [comment, marked as answer]
I don't want to reintroduce the automatic jump that stable has, doesn't feel like a great UX. I'd rather than song select had a visual indicator that there are new (unplayed) beatmaps available, and you could click it to take you to the new beatmap (and perform selection if there was only one, I guess).
